DatadogRUM/Sources/RUMMonitor/Monitor.swift

```diff
@@ -195,6 +195,17 @@ internal class Monitor: RUMCommandSubscriber {
 
 /// Declares `Monitor` conformance to public `RUMMonitorProtocol`.
 extension Monitor: RUMMonitorProtocol {
+    // MARK: - properties
+    var currentSessionID: String? {
+        get {
+            guard let activeSession = self.scopes.activeSession else {
+                return nil
+            }
+
+            return activeSession.sessionUUID.rawValue.uuidString
+        }
+    }
```

Comment:
blocker/ This isn't thread safe and suffers from race condition.
- thread safety: `scopes` are mutated [on monitor's `queue`](https://github.com/DataDog/dd-sdk-ios/blob/7559cdda5ef86a6e3f7294a3c8d2f85a9dfcbdf9/DatadogRUM/Sources/RUMMonitor/Monitor.swift#L134-L138) so we need to sync the getter on that queue as well.
- race condition: `scopes` mutation [is dispatched asynchronously on core's context queue](https://github.com/DataDog/dd-sdk-ios/blob/7559cdda5ef86a6e3f7294a3c8d2f85a9dfcbdf9/DatadogRUM/Sources/RUMMonitor/Monitor.swift#L134) (with `eventWriteContext { ... }`), so session ID is not yet resolved before `process(command:)` returns.

Both problems are not visible in tests this PR adds in `MonitorTests`, because it is using `PassthroughCoreMock` which is single-threaded and we end these tests by calling `monitor.flush()` to force-syncs monitor state. To surface these problems, try adding tests in [RUMMonitorTests.swift](https://github.com/DataDog/dd-sdk-ios/blob/2.5.1/DatadogCore/Tests/Datadog/RUM/) where we use a real instance of the SDK core.

Solving thread-safety is simple as it may only require using `queue.sync {}`. The price would be on performance - it will block the caller thread for the duration of current RUM event processing.

Solving race condition would be more difficult, as it may require sync-ing on core context queue as well. This might be not a good idea, as it will block the caller thread for additional duration of pending core operations. For that reason there is no `sync {}` API provided by core and we may need to use semaphores to workaround it.

Compromise might be guaranteeing thread-safety but explicitly not promising `sessionID` integrity (in API comment). It would mean that following sequence might print the same `sessionID` twice:
```swift
print(monitor.currentSessionID)
monitor.stopSession()
monitor.startView(key: "Foo")
print(monitor.currentSessionID)
```

Alternatively, we can make this API asynchronous:
```swift
monitor.currentSessionID { sessionID in
   print(sessionID) // called asynchronously
}
```
and dispatch `sessionID` in the same model as we are processing RUM commands. This would be the correct solution, but it might be less convenient for users.